src/cookie.rs

```diff
@@ -51,7 +51,7 @@ impl AuthorizationState {
                 // Encode cookie
                 let encoded_cookie = base64engine.encode(encrypted_cookie.as_slice());
 
-                Ok(vec![encoded_cookie, encoded_nonce])
+                Ok(vec![encoded_cookie, encoded_nonce, state.access_token, state.id_token])
```

Comment:
Not sure why I haven't mentioned this earlier, but for a fixed number of return values, a tuple or a struct with named fields would actually be more idiomatic than a `Vec`. That way you can also avoid the `unwrap` at the call site, because the presence of all return values is guaranteed.